tot/data_processing/scaler.py

```diff
@@ -0,0 +1,66 @@
+from dataclasses import dataclass
+
+import numpy as np
+import pandas as pd
+
+
+@dataclass
+class Scaler:
+    transformer: object
+
+    def __post_init__(self):
+        is_transformer_valid = (
+            callable(getattr(self.transformer, "fit", None))
+            and callable(getattr(self.transformer, "transform", None))
+            and callable(getattr(self.transformer, "inverse_transform", None))
+        )
+        if not is_transformer_valid:
+            raise ValueError(
+                "Transformer provided to the Scaler must implement fit, transform and inverse_transform " "methods"
+            )
+
+    def _transform(self, df: pd.DataFrame, fit):
```

Comment:
nice job in adding typing! Can we also type the output? (also for the other functions)